### PR TITLE
Server3

### PR DESCRIPTION
**Pull request by h00dy**

### CODE REVIEW

File: config/dev.exs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+use Mix.Config
+
+config :graphite_limiter,
+run_test_server: false,
+limit: 100,
+sender_pool: 1,
+graphite_url: "http://skystats-dev.qxlint/render/",
```
Comment (benethor) [inline config/dev.exs]: shouldn't we anonymise this?

File: lib/sender.ex [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  defp connect do
+    opts = [:binary, packet: :line, active: false]
+    addr =
+      Application.get_env(:graphite_limiter, :graphite_dest_relay_addr, "localhost")
+      |> String.to_charlist
+    port = Application.get_env(:graphite_limiter, :graphite_dest_relay_port)
+    Logger.debug(fn -> "connecting to: #{addr}:#{port}" end)
+    case :gen_tcp.connect(addr, port, opts) do
+      {:error, _msg} ->
+        Process.sleep(1000)
```
Comment (benethor) [inline lib/sender.ex]: we could probably load the delay value from config

**Review by ch40s3k — APPROVED** [review]

File: vagrant/Vagrantfile [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+apt-add-repository ppa:ansible/ansible
+apt-get -y update
+apt-get -y install ansible
+SCRIPT
+
+Vagrant.configure(2) do |config|
+
+  config.vm.box_download_insecure=true
+  config.vm.box = "ubuntu/xenial64"
+  config.vm.synced_folder "../", "/app/alamo/", create: true
```
Comment (benethor) [inline vagrant/Vagrantfile]: alamo?

**Review by benethor — CHANGES_REQUESTED** [review]

File: lib/graphite_limiter/application.ex [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    else
+      Application.put_env(:graphite_limiter, app_env, String.to_integer(env))
+    end
+  end
+
+  @spec runtime_configuration() :: :ok
+  defp runtime_configuration do
+    set_env("GRAPHITE_FETCH_URL", :graphite_url)
+    set_env("GRAPHITE_QUERY", :graphite_query)
+    set_env("GRAPHITE_FETCH_URL", :graphite_url)
```
Comment (tetianakh) [inline lib/graphite_limiter/application.ex]: Why is `set_env("GRAPHITE_FETCH_URL", :graphite_url)` repeated twice?

File: lib/receiver.ex [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  """
+  @spec accept(non_neg_integer) :: no_return
+  def accept(port) do
+    {:ok, socket} =
+      :gen_tcp.listen(port, [:binary, packet: :line, active: false, reuseaddr: true])
+
+    Logger.info("Accepting connections on port #{port}")
+    # we have to get `sender_pool` env here, as keeping for example it in GraphiteLimiter
+    # gen server state and fetching in while parsing a metric slows down whole process.
+    # Passing it as a arg in function, improves performance significanlty
```
Comment (tetianakh) [inline lib/receiver.ex]: as _an_ arg :)

**Review by tetianakh — APPROVED** [review]

**Review by benethor — APPROVED** [review]

**Pull request merged**